src/iop/colorchecker.c

```diff
@@ -933,25 +931,20 @@ void cleanup_global(dt_iop_module_so_t *module)
   module->data = NULL;
 }
 
-static void picker_callback(GtkWidget *button, gpointer user_data)
+void color_picker_apply(dt_iop_module_t *self, GtkWidget *picker, dt_dev_pixelpipe_iop_t *piece)
 {
-  dt_iop_module_t *self = (dt_iop_module_t *)user_data;
-  if(darktable.gui->reset) return;
-
-  if(self->request_color_pick != DT_REQUEST_COLORPICK_MODULE)
-    self->request_color_pick = DT_REQUEST_COLORPICK_MODULE;
-  else
-    self->request_color_pick = DT_REQUEST_COLORPICK_OFF;
-
-  dt_iop_request_focus(self);
-
   if(self->request_color_pick != DT_REQUEST_COLORPICK_OFF)
   {
     self->gui_update(self);
   }
   dt_control_queue_redraw_center();
+}
 
-  if(self->off) gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(self->off), 1);
+static void _color_picker_callback(GtkWidget *button, gpointer user_data)
+{
+  dt_iop_module_t *self = (dt_iop_module_t *)user_data;
+  if(darktable.gui->reset) return;
```

Comment:
This gets called when _iop_color_picker_callback_button_press (in color_picker_proxy.c) activates the togglebutton. It first sets gui->reset however, so this code returns with setting cst. If it _did_ set cst, the result looks wrong.

Reply:
Sorry, I think I overlooked the `dt_iop_color_picker_set_cst` call... I suppose that the module expects the picked colors in Lab and not Lch so, `dt_iop_color_picker_set_cst(self, iop_cs_Lab)` would be more correct.

As the initialization of the picker_cst is done to the module color space (Lab in this case) when creating the color picker (`dt_color_picker_new` via `_iop_init_picker`), it would be better to remove the call to `dt_iop_color_picker_set_cst`, and so perhaps remove the callback?